src/monit/patch/0002-change_monit_alert_log_error.patch

```diff
@@ -0,0 +1,70 @@
+From 97a5defc6a7fcc6a00f691bb5314ceb8fb7704e9 Mon Sep 17 00:00:00 2001
+From: Abhishek Dosi <abdosi@microsoft.com>
+Date: Mon, 26 Oct 2020 11:40:02 -0700
+Subject: [PATCH] Patch on top of commit Patch is addressing these changes:-
+
+a) Make sure error message is loggged if state is changed to fail first time based on fault tolerance condition
+or we have repeat clause enable for alert.
+
+b) Enable repeat keyword for alert action. Using this we can log
+syslog error message for persistent failures in repeat window.
+
+c) Make sure log and error message are different string so that we do not rate
+limit error message with log message.
+
+Signed-off-by: Abhishek Dosi <abdosi@microsoft.com>
+
+---
+ src/event.c | 6 +++++-
+ src/p.y     | 8 +++++++-
+ 2 files changed, 12 insertions(+), 2 deletions(-)
+
+diff --git a/src/event.c b/src/event.c
+index ed363ee..969568f 100644
+--- a/src/event.c
++++ b/src/event.c
+@@ -336,8 +336,12 @@ static void _handleEvent(Service_T S, Event_T E) {
+                 if (E->state != State_Init || E->state_map & 0x1) {
+                         if (E->state == State_Succeeded || E->state == State_ChangedNot || E->id == Event_Instance || E->id == Event_Action)
+                                 LogInfo("'%s' %s\n", S->name, E->message);
+-                        else
++                        /* Send Error log if state change to failed for 1st time or if we have repeat clause then do periodically */
++                        else if ((E->state_changed) || (E->state == State_Failed && E->action->failed->repeat && E->count % E->action->failed->repeat == 0))
+                                 LogError("'%s' %s\n", S->name, E->message);
++                        else
++                        /* Here we are making log message different so that we do not rate-limit Err and Info in same bucket*/ 
++                                LogInfo("'%s' %s for information\n", S->name, E->message);
```

Comment:
@jleveque should we keep this info message ? I think ERR message should be sufficient . I am not sure we want to send continuous syslog for Info .

Reply:
Can you provide an example of what type of messages will get printed by this clause?

Reply:
@jleveque For example:

Error message (1st time and every repeat cycle if error is persistent)
Oct 26 18:12:54.773398 ERR monit[480]: 'routeCheck' status failed (255) -- no output

Info message (periodic before we hit repeat cycle threshold)
Oct 26 18:12:54.773398 ERR monit[480]: 'routeCheck' status failed (255) -- no output **for information**

Basically Info and error message are same just i have suffix **"for information"**  for info message. The intention was to not to rate-limit ERR message with same pattern as Info message.

I feel we don't need INFO and ERR message should be fine once we update all monit configs to add repeat clause with action so we get periodic ERR messages.

Reply:
@jleveque can you check above comment and provide your feedback

Reply:
> I feel we don't need INFO and ERR message should be fine once we update all monit configs to add repeat clause with action so we get periodic ERR messages.

I agree. However, can you explain under what circumstances this will fall into the INFO logging case?

Reply:
@jleveque basically after first failure case if the repeat keyword is not there in action clause then it will fall into INFO case if the failure is persistent.

Reply:
I see. So as long as we specify the "repeat" keyword, we should never log any INFO messages, anyway, correct?

Reply:
@jleveque Yes that is correct.

Reply:
Then I think we could get rid of the INFO message.

Reply:
@jleveque Thanks Updated.